core/src/memory.rs

```diff
@@ -376,15 +379,26 @@ impl AllocationsContext {
         Ok(start)
     }
 
-    /// Free specific page.
-    ///
-    /// Currently running program should own this page.
+    /// Free specific memory page.
     pub fn free(&mut self, page: WasmPage) -> Result<(), AllocError> {
-        if page < self.static_pages || page >= self.max_pages || !self.allocations.remove(&page) {
-            Err(AllocError::InvalidFree(page.0))
-        } else {
-            Ok(())
+        self.free_range(page..=page)
+    }
+
+    /// Try to free pages in range. Will only return error if range is invalid.
+    ///
+    /// Currently running program should own this pages.
+    pub fn free_range(&mut self, range: RangeInclusive<WasmPage>) -> Result<(), AllocError> {
+        if *range.start() < self.static_pages || *range.end() >= self.max_pages {
```

Comment:
what if start > end?

Reply:
It is checked in `funcs.rs`. But if someone managed to pass 3..=2, no pages will be deallocated because they wouldn't be in the range